src/jobs/jobs.controller.ts

```diff
@@ -390,44 +435,76 @@ export class JobsController {
           throw new HttpException(
             {
               status: HttpStatus.BAD_REQUEST,
-              message: `Invalid new job. User needs to belong to job owner group`,
+              message: `Invalid new job. User needs to belong to job owner group.`,
             },
             HttpStatus.BAD_REQUEST,
           );
         }
         jobInstance.ownerGroup = jobCreateDto.ownerGroup;
       }
     }
-
+    
     if (
       jobConfiguration.create.auth &&
-      jobConfiguration.create.auth in this.jobDatasetAuthorization
+      Object.values(this.jobDatasetAuthorization).includes(jobConfiguration.create.auth)
     ) {
+        // check that jobParams are passed for #dataset jobs
+        if (!jobCreateDto.jobParams) {
+          throw new HttpException(
+            {
+              status: HttpStatus.BAD_REQUEST,
+              message: "Dataset ids list was not provided in jobParams",
+            },
+            HttpStatus.BAD_REQUEST,
+          );
+        }
       // verify that the user meet the requested permissions on the datasets listed
-      const datasetIds = this.checkDatasetIds(jobCreateDto.jobParams);
+      // const datasetIds = await this.checkDatasetIds(jobCreateDto.jobParams);
```

Comment:
Why is this commented out?

Reply:
it's already called in line 392. 
The reasoning behind that was that datasets were only checked when the configuration was set to "#datasetX". Now they are being checked for all configuration if "datasetIds" are passed to jobParams.